lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -735,28 +735,31 @@ def customize_vm(self, vm_obj):
         # Network settings
         adaptermaps = []
         for network in self.params['networks']:
+            guest_map = vim.vm.customization.AdapterMapping()
+            guest_map.adapter = vim.vm.customization.IPSettings()
+
             if 'ip' in network and 'netmask' in network:
-                guest_map = vim.vm.customization.AdapterMapping()
-                guest_map.adapter = vim.vm.customization.IPSettings()
                 guest_map.adapter.ip = vim.vm.customization.FixedIp()
                 guest_map.adapter.ip.ipAddress = str(network['ip'])
                 guest_map.adapter.subnetMask = str(network['netmask'])
-
-                if 'gateway' in network:
-                    guest_map.adapter.gateway = network['gateway']
-
-                # On Windows, DNS domain and DNS servers can be set by network interface
-                # https://pubs.vmware.com/vi3/sdk/ReferenceGuide/vim.vm.customization.IPSettings.html
-                if 'domain' in network:
-                    guest_map.adapter.dnsDomain = network['domain']
-                elif self.params['customization'].get('domain'):
-                    guest_map.adapter.dnsDomain = self.params['customization']['domain']
-                if 'dns_servers' in network:
-                    guest_map.adapter.dnsServerList = network['dns_servers']
-                elif self.params['customization'].get('dns_servers'):
-                    guest_map.adapter.dnsServerList = self.params['customization']['dns_servers']
-
-                adaptermaps.append(guest_map)
+            else:
+                guest_map.adapter.ip = vim.vm.customization.DhcpIpGenerator()
```

Comment:
Please don't assume DHCP if there's no IP information configured. We should have a flag for enabling DHCP.